From 1ab2d020d71a570f691fafbd341782fabb6569d1 Mon Sep 17 00:00:00 2001
From: Zoran Zhan <zoran.zhan@mediatek.com>
Date: Mon, 22 Jan 2024 17:52:05 +0800
Subject: [PATCH] ucm: MediaTek: mt8395-evk: Add HDMIRX config

Add HDMI RX config and modify the priority value
of the capture devices.

Closes: https://github.com/alsa-project/alsa-ucm-conf/pull/389
Signed-off-by: Zoran Zhan <zoran.zhan@mediatek.com>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
Origin: upstream, https://github.com/alsa-project/alsa-ucm-conf/commit/1ab2d020d71a570f691fafbd341782fabb6569d1
Bug: https://bugs.launchpad.net/ubuntu/+source/alsa-ucm-conf/+bug/2051199
---
 ucm2/MediaTek/mt8395-evk/HiFi.conf       | 15 ++++++++++++---
 ucm2/MediaTek/mt8395-evk/mt8395-evk.conf |  1 +
 2 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/ucm2/MediaTek/mt8395-evk/HiFi.conf b/ucm2/MediaTek/mt8395-evk/HiFi.conf
index 81e4423..5c25cfb 100644
--- a/ucm2/MediaTek/mt8395-evk/HiFi.conf
+++ b/ucm2/MediaTek/mt8395-evk/HiFi.conf
@@ -34,6 +34,15 @@ SectionDevice."HDMI2" {
 	]
 }
 
+SectionDevice."HDMI3" {
+	Comment "HDMI RX"
+
+	Value {
+		CapturePriority 200
+		CapturePCM "hw:${CardId},7"
+	}
+}
+
 SectionDevice."Speaker" {
 	Comment "Lineout speaker"
 
@@ -90,7 +99,7 @@ SectionDevice."Headset" {
 	]
 
 	Value {
-		CapturePriority 400
+		CapturePriority 500
 		CaptureChannels 3
 		CapturePCM "hw:${CardId},15"
 	}
@@ -117,7 +126,7 @@ SectionDevice."Mic1" {
 	]
 
 	Value {
-		CapturePriority 300
+		CapturePriority 400
 		CaptureChannels 3
 		CapturePCM "hw:${CardId},15"
 	}
@@ -127,7 +136,7 @@ SectionDevice."Mic2" {
 	Comment "Digital microphone"
 
 	Value {
-		CapturePriority 200
+		CapturePriority 300
 		CaptureChannels 2
 		CapturePCM "hw:${CardId},14"
 	}
diff --git a/ucm2/MediaTek/mt8395-evk/mt8395-evk.conf b/ucm2/MediaTek/mt8395-evk/mt8395-evk.conf
index 1957b69..b739ec0 100644
--- a/ucm2/MediaTek/mt8395-evk/mt8395-evk.conf
+++ b/ucm2/MediaTek/mt8395-evk/mt8395-evk.conf
@@ -18,6 +18,7 @@ BootSequence [
 	cset "name='PGA_3_Mux' AIN3"
 	cset "name='HDMI_OUT_MUX' Connect"
 	cset "name='DPTX_OUT_MUX' Disconnect"
+	cset "name='MULTI_IN1_MUX' HDMI_RX_I2S"
 	cset "name='Lineout Volume' 10"
 	cset "name='Headset Volume' 2"
 	cset "name='PGA1 Volume' 4"
-- 
